disco/extensions/upgrade_simulation/upgrades/common_functions.py

```diff
@@ -749,4 +1076,103 @@ def compare_dict(old, new):
+    return change
+
+
+def create_timepoint_multipliers_dict(timepoint_multipliers):
+    """Creates a dictionary with new load rating, for every property and multiplier.
+    Currently, it only does this for loads. But can be modified to accommodate other elements like PV as well.
+    In raw_dict, value can be accessed as follows:
+    value = raw_dict[property_name][object_name][multiplier_name]
+    
+    In reformatted_dict (which is returned from this function), value can be accessed as follows:
+    value = raw_dict[object_name][property_name][multiplier_name]
+    This value will need to be assigned to the object and run.
+    This hasnt been used yet.
+    
+    Returns
+    -------
+    dict
+    """
+    field = "load_multipliers"
+    if field == "load_multipliers":
+        property_list = ["kW"]
+        object_name = "Load"
+        multiplier_list = []
+        # get combined list of multipliers
+        for key, value in timepoint_multipliers[field].items():
+            multiplier_list = multiplier_list + value
+        df = dss.utils.class_to_dataframe(object_name)
+        df.reset_index(inplace=True)
+        df['name'] = df['index'].str.split(".", expand=True)[1]
+        name_list = list(df['name'].values)
+        del df["index"]
+        df.set_index('name', inplace=True)
+        raw_dict = {}
+        for property in property_list:
+            logger.debug(property)
+            df[property] = df[property].astype(float)
+            new_df = pd.DataFrame(index=name_list, columns=multiplier_list)
+            new_df.index.name = 'name'
+            for multiplier in multiplier_list:
+                logger.debug(multiplier)
+                new_df[multiplier] = df[property] * multiplier
+            raw_dict[property] = new_df.T.to_dict()
+        # reformat dictionary to create desired format
+        reformatted_dict = {}
+        for name in name_list:
+            reformatted_dict[name] = {}
+            for property in property_list:
+                reformatted_dict[name][property] = raw_dict[property][name]
+    return reformatted_dict
+
+
+def apply_timepoint_multipliers_dict(reformatted_dict, multiplier_name, property_list=None, **kwargs):
+    """This uses a dictionary with the format of output received from create_timepoint_multipliers_dict
+    Currently, it only does works loads. But can be modified to accommodate other elements like PV as well.
+
+    In input dict: value can be accessed as follows:
+    value = raw_dict[object_name][property_name][multiplier_name]
+    In this function, value will be assigned to corresponding property and run.
+    This hasnt been used yet.
+    
+    Returns
+    -------
+    dict
+    """
+    field = "load_multipliers"
+    name_list = list(reformatted_dict.keys())
+    if property_list is None:
+        property_list = list(reformatted_dict[name_list[0]].keys())
+    if field == "load_multipliers":
+        for name in name_list:
+            dss.Loads.Name(name)

```

Comment:
This is a very slow way of iterating. It is better to use the native OpenDSS iteration. Could you flip it easily?